packages/headless/src/controllers/generated-answer/headless-generated-answer.ts

```diff
@@ -196,110 +30,38 @@ export function buildGeneratedAnswer(
   engine: SearchEngine,
   props: GeneratedAnswerProps = {}
 ): GeneratedAnswer {
-  if (!loadGeneratedAnswerReducer(engine)) {
-    throw loadReducerError;
-  }
-
   const {dispatch} = engine;
-  const controller = buildController(engine);
-  const getState = () => engine.state;
-
-  const isVisible = props.initialState?.isVisible;
-  if (isVisible !== undefined) {
-    dispatch(setIsVisible(isVisible));
-  }
-  const initialResponseFormat = props.initialState?.responseFormat;
-  if (initialResponseFormat) {
-    dispatch(updateResponseFormat(initialResponseFormat));
-  }
-
-  const fieldsToIncludeInCitations = props.fieldsToIncludeInCitations;
-  if (fieldsToIncludeInCitations) {
-    dispatch(registerFieldsToIncludeInCitations(fieldsToIncludeInCitations));
-  }
-
-  subscribeStateManager.subscribeToSearchRequests(engine);
+  const controller = buildCoreGeneratedAnswer(
+    engine,
+    generatedAnswerAnalyticsClient,
+    props
+  );
 
   return {
     ...controller,
 
     get state() {
-      return getState().generatedAnswer;
+      return controller.state;
     },
 
     retry() {
-      dispatch(executeSearch({legacy: logRetryGeneratedAnswer()}));
-    },
-
-    like() {
-      if (!this.state.liked) {
-        dispatch(likeGeneratedAnswer());
-        dispatch(logLikeGeneratedAnswer());
-      }
-    },
-
-    dislike() {
-      if (!this.state.disliked) {
-        dispatch(dislikeGeneratedAnswer());
-        dispatch(logDislikeGeneratedAnswer());
-      }
-    },
-
-    openFeedbackModal() {
-      dispatch(openGeneratedAnswerFeedbackModal());
-    },
-
-    closeFeedbackModal() {
-      dispatch(closeGeneratedAnswerFeedbackModal());
-    },
-
-    sendFeedback(feedback) {
-      dispatch(logGeneratedAnswerFeedback(feedback));
-      dispatch(sendGeneratedAnswerFeedback());
-    },
-
-    sendDetailedFeedback(details) {
-      dispatch(logGeneratedAnswerDetailedFeedback(details));
-      dispatch(sendGeneratedAnswerFeedback());
-    },
-
-    logCitationClick(citationId: string) {
-      dispatch(logOpenGeneratedAnswerSource(citationId));
-    },
-
-    logCitationHover(citationId: string, citationHoverTimeMs: number) {
-      dispatch(logHoverCitation(citationId, citationHoverTimeMs));
+      dispatch(
+        executeSearch({
+          legacy: generatedAnswerAnalyticsClient.logRetryGeneratedAnswer(),
+        })
+      );
     },
 
     rephrase(responseFormat: GeneratedResponseFormat) {
       dispatch(updateResponseFormat(responseFormat));
```

Comment:
Wouldn't you end up doing 
`controller.rephrase(responseFormat);`
instead of redefining here the same thing as in the CoreController?

Right now this action exists on the Core Controller, but if we want to add behavior to it, that would impact all controllers dependent on the Core, right now it would have no effect?

Reply:
Sure thing!